File: jansetu-ai/core/pipeline.ts
import { detectLanguage, translateToEnglish } from "./translator.js";
import { classifyComplaint } from "./classifier.js";
import { scoreUrgency } from "./urgency.js";
import { generateSummary } from "./summary.js";

export interface ProcessedComplaint {
  translated_text: string;
  language: string;
  category: string;
  department: string;
  urgency_level: string;
  confidence_score: number;
  is_duplicate: boolean;
  duplicate_of: string | null;
  similarity_score: number | null;
  summary: string;
}

export function processComplaint(
  rawText: string,
  language?: string,
  location?: string
): ProcessedComplaint {
  const lang = language ?? detectLanguage(rawText);
  const translated = translateToEnglish(rawText, lang);
  const { category, department, confidence } = classifyComplaint(translated);
  const urgency = scoreUrgency(translated);
  const summary = generateSummary(translated, category, department, urgency, location);
  return {
    translated_text: translated,
    language: lang,
    category,
    department,
    urgency_level: urgency,
    confidence_score: confidence,
    is_duplicate: false,
    duplicate_of: null,
    similarity_score: null,
    summary,
  };
}

File: jansetu-ai/backend/ai/pipeline.py
from ai.translator import detect_language, translate_to_english
from ai.classifier import classify_complaint, get_department
from ai.urgency import score_urgency
from ai.summary import generate_summary

def process_complaint(raw_text, language=None, location=None):
    if language is None:
        language = detect_language(raw_text)
    translated = translate_to_english(raw_text, language)
    category, confidence = classify_complaint(translated)
    department = get_department(category)
    urgency = score_urgency(translated, category)
    summary = generate_summary(translated, category, department, urgency, location)
    return {
        "translated_text": translated, "language": language,
        "category": category, "department": department,
        "urgency_level": urgency, "confidence_score": confidence,
        "is_duplicate": False, "duplicate_of": None,
        "similarity_score": None, "summary": summary
    }
